src/ImageSharp/Processing/Processors/Convolution/ReadOnlyKernel.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace SixLabors.ImageSharp.Processing.Processors.Convolution
+{
+    /// <summary>
+    /// A stack only, readonly, kernel matrix that can be indexed without
+    /// bounds checks when compiled in release mode.
+    /// </summary>
+    internal readonly ref struct ReadOnlyKernel
```

Comment:
Also to get back to my point above with respect to clarity, imho this should be called `ReadOnlyKernel2D`.

Reply:
See above, there's only be one type.